pytorch_lightning/utilities/auto_restart.py

```diff
@@ -737,3 +746,94 @@ def _teardown_dataloader_get_iterators() -> None:
     if get_iterator:
         DataLoader._get_iterator = get_iterator
         del DataLoader._ori_get_iterator
+
+
+def _validate_iterable_dataset(dataloader: DataLoader) -> Tuple[bool, str]:
+
+    SUPPORTED_SAMPLERS = (RandomSampler, SequentialSampler, DistributedSampler)
+
+    dataset = dataloader.dataset
+
+    if getattr(dataset, "__next__", None) is None:
+        raise MisconfigurationException(
+            "Fault Tolerant Training doesn't support an IterableDataset without a `__next__` "
+            "method implemented. Hint: We recommend you to move your logic inside and rely "
+            "on a sampler, generator to perform the iteration."
+        )
+
+    samplers = {k: v for k, v in dataset.__dict__.items() if isinstance(v, Sampler)}
+
+    if not samplers:
+        raise MisconfigurationException(
```

Comment:
this would then be either an attribute or a value error

Reply:
Done !